src/java.base/share/classes/jdk/internal/foreign/abi/aarch64/CallArranger.java

```diff
@@ -266,7 +266,8 @@ Homogeneous float aggregates (HFAs) can be copied in a field-wise manner, i.e. t
         StructStorage[] structStorages(GroupLayout layout, boolean forHFA) {
             int regType = forHFA ? StorageType.VECTOR : StorageType.INTEGER;
             int numChunks = (int)Utils.alignUp(layout.byteSize(), MAX_COPY_SIZE) / MAX_COPY_SIZE;
-            int requiredStorages = forHFA ? layout.memberLayouts().size() : numChunks;
+            List<MemoryLayout> scalarLayouts = forHFA ? TypeClass.scalarLayouts(layout) : null;
```

Comment:
Perhaps, instead of two conditionals, it would be better to use a simple `if` ? I'm a bit worried about one path being `null` which then works in the subsequent conditional because the leading condition is the same. Seems a bit brittle.

Reply:
That's also a possibility, I can put the initialization of `regType` into the `if` as well. I'll do that.